### PR TITLE
POS Navigation and account authentication

### PR DESCRIPTION
**Pull request by gitnamehere**
This took me wayyy too long

This PR adds in navigational buttons in the POS, adds and modify tables, and adds account authentication

(styling can be fixed later in a different pr)

Testing:
* make a new copy of `env.json` from `sample-env.json` and fill in relevant informtation
* `npm i`
* `npm run setup-demo`
* `npm start`
* open a new terminal and log in to the database with psql
* \c into the pos database, and type in `SELECT * FROM accounts`;
* verify the account with the credentials in the env file is inserted (the password will be hashed)
* go to http://localhost:3000/pos/ordering/
* see that a redirect page is displayed, click on the redirect link
* log in based on the provided credentials in the env.json file, notably boss_user and boss_password
* verify that login works and you can access the pos as normal
* click on the logout button in the POS home screen on the top right (don't mind the styling)
* verify that logging out works, and it redirects you to the login screen
* log back in, and go to http://localhost:3000/pos/ordering/
* test out the navigational buttons in the navbar, and verify that it works 
* go to http://localhost:3000/pos/item-management/create.html
* test out the navigational buttons, and verify that it works normally
* go to http://localhost:3000/pos
* test out that some of the buttons navigates you correctly
* restart the node server
* verify that you are still authenticated

time to sleep

### CODE REVIEW

**Review by nhokzminh698 — APPROVED** [review]
I still authenticate after rerun the node, everything works when following the instruction!! Account authentication go wild. U tell em!!

**Comment by wetefok** [comment]
After creating the boss account for the very first time, my authentication was not maintained between different tabs. I would be logged into the POS, but if I were to open another POS tab, it would redirect me to the login page. This issue did not persist after terminating the node server and restarting it. I wanted to create a brand new boss account to retest by changing the `boss_user` and `boss_password` values in `env.json` but it did not create a new account after running npm run setup-demo.

File: app/frontend/public/pos/login/script.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        }),
+    })
+        .then((response) => {
+            // redirect to homepage if everything checks out
+            response.status === 200
+                ? (window.location = "/pos")
+                : alert("Invalid Username or Password");
+        })
+        .catch((error) => console.log(error));
+});
```
Comment (wetefok) [inline app/frontend/public/pos/login/script.js:30]: We ending files without whitespace Jason? 🤨

File: app/frontend/private/pos/orders/styles.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    align-items: center;
+    height: 100%;
+    width: auto;
+}
+
+.header-button {
+    margin-left: 16px;
+    font-size: 24px;
+    font-weight: 700;
+}
```
Comment (wetefok) [inline app/frontend/private/pos/orders/styles.css:32]: We ending files without whitespace Jason? 🤨

File: app/frontend/private/pos/category-management/styles.css [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    align-items: center;
+    height: 100%;
+    width: auto;
+}
+
+.header-button {
+    margin-left: 16px;
+    font-size: 24px;
+    font-weight: 700;
+}
```
Comment (wetefok) [inline app/frontend/private/pos/category-management/styles.css:28]: We ending files without whitespace Jason? 🤨

File: app/frontend/public/pos/login/styles.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+.card {
+    margin: auto;
+    padding: 32px;
+}
+
+input,
+button {
+    width: 400px;
+}
```
Comment (wetefok) [inline app/frontend/public/pos/login/styles.css:19]: We ending files without whitespace Jason? 🤨

File: app/frontend/public/pos/login/index.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                        id="password"
+                        placeholder="Password"
+                    />
+                </div>
+                <button id="submit" class="btn btn-primary">Login</button>
+            </div>
+        </div>
+        <script src="script.js"></script>
+    </body>
+</html>
```
[the remark this answers is not in the document]
Comment (gitnamehere) [reply app/frontend/public/pos/login/index.html:34]: ???

File: app/routes/api.js [hunk 6]
```diff
@@ -27,15 +57,24 @@ function query(query, values, res, sendStatus = false) {
...
+    pool.query("SELECT * FROM tokens WHERE token = $1", [posAuth])
+        .then((result) => {
+            if (result.rows.length === 0) {
+                // redirect to login page
+                // https://expressjs.com/en/4x/api.html#res.redirect
+                return res.redirect("/pos/login");
+            }
+
+            // not putting next() in here caused me an hour of pain
+            next();
```
Comment (wetefok) [inline app/routes/api.js:72]: what does next do

**Review by wetefok — CHANGES_REQUESTED** [review]
Tsk Tsk Tsk. Naughty Jason.

File: app/frontend/public/pos/login/script.js (hunk 1 above)
Comment (gitnamehere) [reply app/frontend/public/pos/login/script.js:30]: ???

File: app/frontend/private/pos/orders/styles.css (hunk 2 above)
Comment (gitnamehere) [reply app/frontend/private/pos/orders/styles.css:32]: ???

File: app/frontend/private/pos/category-management/styles.css (hunk 3 above)
Comment (gitnamehere) [reply app/frontend/private/pos/category-management/styles.css:28]: ???

File: app/frontend/public/pos/login/styles.css (hunk 4 above)
Comment (gitnamehere) [reply app/frontend/public/pos/login/styles.css:19]: ???

File: app/routes/api.js (hunk 6 above)
Comment (gitnamehere) [reply app/routes/api.js:72]: pass it off to the request handler or middleware

File: app/frontend/public/pos/login/index.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (nhokzminh698) [reply app/frontend/public/pos/login/index.html:34]: I find html and css white space doesnt work if I made it and save it just get rid of the white space but I think prettier fixed it.

**Review by wetefok — APPROVED** [review]
jason

**Pull request merged**